### ISSUE TITLE
Can't make it work with i18n

### ISSUE THREAD

**Issue by MaksJS** [techpines/asset-rack#42]
Hi everyone,

I'm using asset-rack to serve Jade assets for my Backbone app :

``` coffeescript

i18n.configure
  locales  : ['en']
  register : global
  directory: 'config/locales'
  extension: '.json'

app.configure ->
  app.use i18n.init
  app.use new JadeAsset
    url    : '/backbone/templates.js'
    dirname: __dirname + '/app/assets/javascripts/app/templates'
```

Well it works perfectly but I can't use my i18n module as I did before in my Jade templates

``` jade
b #{__('foo')}
```

Throws __ undefined. I'm using this module : https://github.com/mashpie/i18n-node

Any solution ?
Thanks :)

**Comment by techpines** [comment]
You need to have that i18n code run on the client side.

That `__` is evaluated in the browser.

**Comment by MaksJS** [comment]
There's no client library with this module unfortunately. I need a way to evaluate i18n strings before serving Jade asset to the client or move to another i18n module lol

**Comment by techpines** [comment]
That's a bummer.  Let me think on this one.  Basically, you need to do the compilation before, and then pull down the correct "templates" based on the language. 

Otherwise, it's got to be client based.

**Comment by MaksJS** [comment]
I've found a solution ! That's so tricky and hard to parse but it works, without client library (so it's still pre-compiled)
In JadeAsset's class, createContents method : 

``` coffeescript
        for fileObject in @fileObjects
            compiled = fileObject.compiled.toString()
            _compiled = fileObject.compiled.toString()
            _compiled = _compiled.replace /\+ escape/g, '+ \n escape'
            i18nStrings = _compiled.match /escape\(.+(__\('.+'\))\)/g
            if i18nStrings?
              for i18nString in i18nStrings
                i18nString = i18nString.match(/escape\(.+(__\('.+'\))\)/)[1]
                _i18nString = i18nString.match(/__\('(.+)'\)/)[1]
                word = "'#{__.call(scope:{locale:'en'}, _i18nString)}'"
                compiled = compiled.substring(0, compiled.indexOf(i18nString)) + word + compiled.substring(i18nString.length + compiled.indexOf(i18nString), compiled.length)
            @contents += "'#{fileObject.funcName}': #{compiled},"
```

Maybe we could had beforeCompile and afterCompile options to the constructor ? Then i could externalize this logic to a module and pass it to beforeCompile option.

I know people who would be happy to use i18n for Jade on the client : https://github.com/mashpie/i18n-node/issues/31 :)

**Comment by techpines** [comment]
Cool, I knew this was not an easy problem, but you're solution looks pretty good!

If you make a pull request for `beforeCompile`, I'll merge it right away.

Cheers.

**Comment by MaksJS** [comment]
Well you're right I should try to parse this before compiling cause it's easier to deal with a Jade template than a Javascript compiled template. I'll rewrite my parser accordingly :)
Thanks for your help !

**Comment by MaksJS** [comment]
It's gonna be a lot easier now

``` coffeescript
  app.use new JadeAsset
    url: '/backbone/templates.js'
    dirname: __dirname + '/app/assets/javascripts/app/templates'
    clientVariable: 'JST'
    beforeCompile: (content) -> content.replace /#\{__\('(.+)'\)\}/g, (match, str) -> __.call scope:{locale:'fr'}, str
```

And then you'll get

``` jade
b Administrateur
```

instead of

``` jade
b #{__('Administrator')}
```
